plugin/src/main/java/io/jenkins/plugins/analysis/core/model/JobAction.java

```diff
@@ -202,6 +206,20 @@ public boolean isTrendVisible() {
         return trendChartType != TrendChartType.NONE && createBuildHistory().hasMultipleResults();
     }
 
+    public boolean noWarnings(){
+        History results =createBuildHistory(); //Get past build history
+        if(results instanceof AnalysisHistory){
```

Comment:
This check should not be necessary

Reply:
As far as I understand, createBuildHistory() could return two types of objects: NullAnalysisHistory or AnalysisHistory. When NullAnalysisHistory is returned, it means there is no build history, so we are going to return true directly. If it's AnalysisHistory, we will start iterating through it.

Reply:
But this is the clue of the `NullObjectPattern`, you really do not need to care: your algorithm will terminate with `true` anyway since there are no build results.